context_internal_test.go

```diff
@@ -55,3 +55,12 @@ func TestCurrentSpan(t *testing.T) {
 	require.NotNil(t, span, "CurrentSpan() should always return something")
 	assert.NotEqual(t, uint64(0), span.TraceID(), "mock tracer is now Zipkin-compatible")
 }
+
+func TestContextWithoutHeaders(t *testing.T) {
+	ctx := WrapWithHeaders(context.Background(), map[string]string{"k1": "v1"})
+	assert.Equal(t, map[string]string{"k1": "v1"}, ctx.Headers())
+	ctx2 := WithoutHeaders(ctx)
+	assert.Nil(t, ctx2.Value(contextKeyHeaders))
+	_, ok := ctx2.(ContextWithHeaders)
+	assert.False(t, ok)
```

Comment:
We should test that we're also hiding `contextKeyTChannel`. I think we can do so by using `NewContextBuilder` and `SetShardKey`, then testing that we're hiding the shard key.